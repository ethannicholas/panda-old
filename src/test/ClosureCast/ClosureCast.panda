def addI:(Int, Int)=>(Int) := Int.+
def addO:(Int, Int)=>(Object) := Int.+
Console.writeLine(addI(3, 5))
Console.writeLine(addO(3, 5))

def writeO := method(o:Object) {
    Console.writeLine("Object: \{o}")
}
def writeI:(Int)=>() := writeO
writeO("Object!")
writeI(45)
