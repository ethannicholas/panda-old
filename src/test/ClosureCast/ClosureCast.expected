8
8
Object: Object!
Object: 45
